### DISCUSSION TITLE
Is there interpolation support?

### DISCUSSION THREAD

**Discussion by alterae** [jcornaz/heron#147] [Q&A] [answered]
I'm currently making a simple pong game using Heron, and while it mostly works there is one issue.

When the ball moves at a sufficiently high speed, it goes *through* the paddles.

Is there any way to enable physics interpolation or otherwise prevent this?

**Comment by jcornaz** [comment, marked as answer]
Hi, 

This is a common issue with discrete physics simulation, and it has no real easy answer.

But here are the two things you may try:

* Increase the number of steps per second (by configuring the [PhysicsSteps](https://docs.rs/heron/0.11.1/heron/struct.PhysicsSteps.html) resource). But also make sure to also have a high frame rate (which shouldn't be a problem for a pong)

* Play with rapier CCD (Continuous Collision Detection) feature: https://rapier.rs/docs/user_guides/javascript/rigid_bodies/#continuous-collision-detection (The rapier `IntegrationParameters` is a resource)

All of that being said, there is maybe something we we can do on heron to aleviate this kind of problem. Maybe we have to re-introduce `PhysicsStep` mode that would go at a fixed rate, desynchronized from the bevy update. Heron had that before, but was removed because it also comes with many problems and complication (see #107). But if it isn't the default, it may be helpful to have an opt-in for that.

**Reply by alterae** [reply]
I'll look into rapier's ccd.  thank you
